src/Core/Notifications/INotificationService.cs

```diff
@@ -86,7 +74,7 @@ private void CurrentInfoBar_ButtonClick(object sender, InfoBarButtonClickedEvent
         {
             try
             {
-                var matchingAction = activeNotification.Item2.Actions.FirstOrDefault(x => x.CommandText == e.ClickedButtonText);
+                var matchingAction = activeNotification?.Item2.Actions.FirstOrDefault(x => x.CommandText == e.ClickedButtonText);
```

Comment:
I was thinking of a scenario where the user would click on a button and before the callback is scheduled, we replace the infobar with a new notification.
I don't think it can happen since both will be on the UI thread, so the callback must be executed before we can access the UI thread to replace the infobar, and while we're replacing, the thread is blocked so the user cannot click. 
But I'm paranoid so I thought it might be worth adding the check here anyway...

Reply:
```
I don't think it can happen since both will be on the UI thread, so the callback must be executed before we can access the UI thread to replace the infobar
```

In general, it is possible for a thread to [temporarily yield control of the UI thread](https://github.com/microsoft/vs-threading/blob/main/doc/cookbook_vs.md#how-to-switch-to-or-use-the-ui-thread-with-a-specific-priority), which would allow other code to execute on the UI thread. In this case, our code doesn't yield control, but we don't know what happens inside any VS-code we call.

However, I don't think we need to worry about it at the moment. So far, we've only thought about displaying a single notification at a time, and there is only one possible notification we could show. To make the service more general, we would need to think the UX for displaying more than one notification at a time. Locking might well be part of that.

Reply:
Even for a single notification, it can be triggered by multiple analysis failures. (which gives me an idea for another task: not to display a notification of a notification of same id is currently shown)

Reply:
Yep, true.

Not displaying the same notification if it already being shown: there is already a more restrictive requirement in the spec document: _Only one instance of the notification should be shown per VS session._ Implementing that will require thread safety/state management.

You could either commit this change as is and tweak the locking as necessary when implementing "one notification per session", or add locking as part of that feature - do whichever is easiest for you.